src/x/context/context.go

```diff
@@ -340,6 +359,9 @@ func (c *ctx) StartSampledTraceSpan(name string) (Context, opentracing.Span, boo
 
 	child := c.newChildContext()
 	child.SetGoContext(childGoCtx)
+	if child.DistanceFromRootContext() == maxDistanceFromRootContext {
```

Comment:
why do we only want an equality check here and not `>=`?

Reply:
This is done to signify we only mark a single span with an error (the 100th one). Starting at 100th and any deeper, we will not permit to create additional child spans and return a no-op: it's happening in the beginning of the method.